services/QuillLMS/spec/controllers/teachers/progress_reports/diagnostic_reports_controller_spec.rb

```diff
@@ -19,12 +19,72 @@
       let!(:activity_session) { create(:activity_session, classroom_unit: classroom_unit, activity: activity, user: student) }
 
       it "redirects to the correct page" do
-        get :report_from_classroom_unit_activity_and_user, params: ({classroom_unit_id: classroom_unit.id, user_id: student.id, activity_id: activity.id})
+        get :report_from_classroom_unit_and_activity_and_user, params: ({classroom_unit_id: classroom_unit.id, user_id: student.id, activity_id: activity.id})
         expect(response).to redirect_to("/teachers/progress_reports/diagnostic_reports#/u/#{unit.id}/a/#{activity.id}/c/#{classroom.id}/student_report/#{student.id}")
       end
     end
   end
 
+  describe '#redirect_to_report_for_most_recent_activity_session_associated_with_activity_and_unit' do
+  describe 'when the activity is a diagnostic' do
```

Comment:
Very tiny detail, but it looks like you're missing a level of indentation here which leads to the `end` on line 86 being under-indented.  This doesn't break the tests, obviously, but does make it a bit harder to read.